libs/block-template-custom-element/package.json

```diff
@@ -57,13 +58,23 @@
     "displayName": "Template",
     "examples": [
       {
-        "name": "World"
+        "https://blockprotocol.org/@blockprotocol/types/property-type/name/": "World"
       }
     ],
     "icon": "public/omega.svg",
     "image": "public/block-preview.png",
     "name": "block-template-custom-element",
     "protocol": "0.3",
-    "schema": "https://blockprotocol.org/@blockprotocol/types/entity-type/thing/v/2"
```

Comment:
> have to update the block's schema URL in two places, which we should avoid.

This was my main motivation for the current format ([see this comment](https://github.com/blockprotocol/blockprotocol/pull/1158#discussion_r1132987188)), so I feel fairly strongly about this too.

> I am instinctively attracted to (1) because it is simpler for the block use case where you don't need anything but the block + immediate subgraph generated (essentially operating as it does now, perhaps with a file name change), and specifying the more complicated codegen object is for use cases where you need multiple types generated and want to specify filenames and so on.

I actually disagree with this, I think two competing formats and needing to understand how to migrate from one to the other (or use both at once?) is more confusing and _less_ simple. We expect the vast majority of blocks to be created from the template (or following some future tutorials), which will explicitly give users the `codegen` boilerplate, so I think the simpler block cases boil down to "change the only URL you see in this object to your URL" and it's functionally the same difficulty as having `schema`.

As such I'd prefer if we go with this:

 > We drop this key and take the value for schema in block-metadata.json from whatever entry in codegen is marked as blockEntity: true.

To be able to do that I think we'd need to stop what I've done at the moment where I'm allowing a `blockEntity: true` _per_ file. My intentions there was that technically this codegen doesn't need to only be ran for a single block, one could define one codegen for a whole suite of blocks for example (using the file mapping would make that fairly easy). However, I don't see a huge benefit for that, and I guess we can continue to allow that in codegen, but just disallow it when parsing a config from a block's `package.json`. If you're happy with that I'll go and implement this

Reply:
Yes, sounds good. Can you update the `build` script at the same time to populate `schema` from the new place please?

Reply:
I've applied a constraint of only accepting one `blockEntity` definition in the `package.json.`
[Only allow one blockEntity within block's codegen parameters](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/10d6f99a865411c21ccbffd97b58157271ef3397)

You can checkout [Add demonstration of multiple codegen blockEntity defs](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/7bd695aa341c291c91f3124e38119fba64b16f58) to see it working: <img width="1821" alt="image" src="https://user-images.githubusercontent.com/25749103/225031793-ffa075bb-6227-46d7-90fb-4679ead90d6b.png">


> This is the block's declared schema which is used as the value for schema in block-metadata.json, which is generated by block-scripts/build.
> [...]
> Yes, sounds good. Can you update the build script at the same time to populate schema from the new place please?

Was a bit unsure of how to do this as `generate-dist-block-metadata.js` seems to just set `schema: "block-schema.json"`, but I'm guessing that's implicitly overridden by `...blockprotocol`.

[Use blockEntity type URL from codegen object in block-metadata](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/38b5246b58a055a8703679c9394e9476df63cca8)

Seems to work:
<img width="1043" alt="image" src="https://user-images.githubusercontent.com/25749103/225035192-d35257f4-f485-4c5c-8a91-58674222f5f8.png">